### ISSUE TITLE
[Bug] Attribute tables vector layers in Natural Earth sample dataset not readable

### ISSUE THREAD

**Issue by ecodiv** [OSGeo/grass#4172]
## Describe the bug
The attribute tables of the vector layers that come with the Natural Earth Sample dataset that you can install from within GRASS GIS are not readable. Trying to open gives an error message (see screenshot below).  After dismissing the error message, an empty attribute table opens. The same is true if you download (presumably the same) dataset from the website and for the full world dataset.

## To reproduce
1. Open GRASS GIS, go to the data panel
2. Select the Download Sample Project option
3. Select the Natural Earth sample dataset and download
4. Open any of the vector layers
5. Open the attribute table of the vector layer

## Screenshots

![image](https://github.com/user-attachments/assets/2ae63d4b-3d90-4624-8401-d0d6c8904307)

## System description

- Operating System: Ubuntu 24.04
- GRASS GIS version: 
  version=8.4.0dev
  date=2024
  revision=aa1ba17ef8
  build_date=2024-07-06
  build_platform=x86_64-pc-linux-gnu
  build_off_t_size=8
  libgis_revision=bb40fdd5a0
  libgis_date=2024-06-18T13:26:55+00:00
  proj=9.4.0
  gdal=3.8.4
  geos=3.12.1
  sqlite=3.45.1
- Python version: 3.12.3 (main, Jul 31 2024, 17:43:48) [GCC 13.2.0]
- Wx: 4.2.1 gtk3 (phoenix) 
-  wxWidgets 3.2.4

## Additional context
Reporting here because it influences the experience of grass gis, but perhaps better to report in the grass-website repository?

**Comment by ecodiv** [comment]
The problem might be with a dbf backend being used. I have created a sample data set with the same layers as the Natural Earth full data set, using sqlite as backend, and that works without problem (you can download it from https://ecodiv.earth/share/natural_earth_dataset.zip). Not sure if the problem is with my setup, but irrespectively, is there a reason to use a dbf backend instead of sqlite?

**Comment by veroandreo** [comment]
> The problem might be with a dbf backend being used. I have created a sample data set with the same layers as the Natural Earth full data set, using sqlite as backend, and that works without problem for me (you can download it from https://ecodiv.earth/share/natural_earth_dataset.zip). Not sure if the problem is with my setup, but irrespectively, is there a reason to use a dbf backend instead of sqlite?

Yes, the issue is dbf as the default backend in that project, it's an old one. Can we use your modified version to upload to the website? @neteler what do you think?

**Comment by neteler** [comment]
Since the procedure (https://grasswiki.osgeo.org/wiki/Convert_all_GRASS_6_vector_maps_to_GRASS_7) is quite some work, using the donated updated version by @ecodiv would be very welcome! Was this procedure used?

In wxgui, the URL is stored here:
https://github.com/OSGeo/grass/blob/cb013cccc40978c7db9f1c4ace7924263d00a6db/gui/wxpython/startup/locdownload.py#L75

which points to https://zenodo.org/record/3968936/files/natural-earth-dataset.tar.gz

In my view, the correct way to update would be
- change the code above to point to the "all versions" DOI: https://doi.org/10.5281/zenodo.3762773
- notify @baharmon to update the Zenodo record with the file as a new version, file provided by @ecodiv 
- backport to G84.
- done

**Comment by ecodiv** [comment]
> Since the procedure (https://grasswiki.osgeo.org/wiki/Convert_all_GRASS_6_vector_maps_to_GRASS_7) is quite some work, using the donated updated version by @ecodiv would be very welcome! Was this procedure used?
> 

Hi @neteler no, I downloaded the data anew. However, I have now created an updated dataset using the steps as outlined in the link you provided. 

- https://ecodiv.earth/share/natural_earth_dataset.zip
- https://ecodiv.earth/share/natural-earth-dataset.tar.xz

**Comment by ecodiv** [comment]
@neteler @veroandreo @baharmon And below the link to the updated global_dataset dataset. Note that I replaced two data layers, roads and railroads with newly downloaded and imported Natural Earth layers, because the one in the original global data set did not contain attribute tables. 

- https://ecodiv.earth/share/global-dataset.zip

**Comment by neteler** [comment]
I like it much better to have the dataset updated, too :-) Thanks @ecodiv .

**Comment by neteler** [comment]
> However, I have now created an updated dataset using the steps as outlined in the link you provided.
> 
>     * https://ecodiv.earth/share/natural_earth_dataset.zip
> 
>     * https://ecodiv.earth/share/natural-earth-dataset.tar.xz

May I suggest to add a version into the file names to avoid confusion which one is based on the old data and which one contains fresh Natural Earth data?

**Comment by ecodiv** [comment]
OK, let me create for both datasets new versions with all data layers based on what is currently available on Natural Earth, that makes it less confusing. Will do that today or tomorrow. @baharmon, as this would be a new version, would it be OK if I add one or two extra layers to the global data set?

**Comment by ecodiv** [comment]
@baharmon @veroandreo @neteler I recreated the data sets (global data set and natural_earth_dataset) with the latest layers, and with the following changes:

- Replaced the landuse map 2015 for the landuse map 2022
- Added the Explorer Base Map raster layer
- I had to recreate the ocean layer because the original ocean layer could not be imported correctly.
- I added a markdown file with for each layer a brief description of the data source

The links to the data sets:

- http://ecodiv.earth/share/global-dataset.zip
- http://ecodiv.earth/share/natural_earth_dataset.zip

**Comment by baharmon** [comment]
@neteler @veroandreo @ecodiv I have uploaded both datasets to Zenodo and added @ecodiv as a contributor, but have not yet published them. Please let me know if you would like to review entries. Perhaps we should add GRASS Development team as a contributor too? Here are previews of the 2.0.0 versions of the [Natural Earth Dataset](https://zenodo.org/records/13370131?preview=1&token=eyJhbGciOiJIUzUxMiJ9.eyJpZCI6IjZhOGMxYWE4LTE4NzAtNDRkMS04OGUxLTRkZjVkZjljMjcyOSIsImRhdGEiOnt9LCJyYW5kb20iOiIyYmNiNjM0MTExY2M4MzcwZDVkMTk0YmRkZTdiNDdiMiJ9.dEZSLfedwAjG3Q5RIHl7te1G4f5DjJzvh14m979VRdlDm1Kyr4N83pmbHKYzNzQhBFacfME-WYJk-hk5tFDncg) and the [Global Dataset](https://zenodo.org/records/13370326?preview=1&token=eyJhbGciOiJIUzUxMiJ9.eyJpZCI6ImU4ZGQ4ZmJlLWVjY2ItNDJhNy05NDRhLWQ0ODhkYjAzNTNkMiIsImRhdGEiOnt9LCJyYW5kb20iOiIyNjU3NGE0NDdmOTljMzYwYzIwZjNhM2IyOWFjNWRjMiJ9.FRyLNaq31Z_3YRm20GvKv2-Q1VGFfK5r2ccZY-ZoeOEKYr6sjL6WVZ-ay1WKoGSfsJR7gmCRu_KpelKo8LBOlA).

**Comment by ecodiv** [comment]
> @neteler @veroandreo Perhaps we should add GRASS Development team as a contributor too? 

Good idea

**Comment by baharmon** [comment]
@neteler @veroandreo @ecodiv Published. Here are the DOIs. Theses are the DOIs for all records which will resolve to the latest release. Will open an issue on grass-website to update the global dataset links. 

* Natural Earth Dataset for GRASS GIS: [10.5281/zenodo.3762773](https://zenodo.org/doi/10.5281/zenodo.3762773)
* Global Dataset for GRASS GIS: [10.5281/zenodo.3359631](https://zenodo.org/doi/10.5281/zenodo.3359631)
